include/sys_matrix.h

```diff
@@ -0,0 +1,71 @@
+#ifndef SYS_MATRIX_H
+#define SYS_MATRIX_H
+
+#include "z64math.h"
+
+struct GraphicsContext;
+struct GameState;
+
+typedef enum {
+    /* 0 */ MTXMODE_NEW,  // generates a new matrix
+    /* 1 */ MTXMODE_APPLY // applies transformation to the current matrix
+} MatrixMode;
+
+// extern Mtx gMtxClear;
+// extern MtxF gMtxFClear;
+
+/* Stack operations */
+
+void Matrix_Init(struct GameState* gameState);
+void Matrix_Push(void);
+void Matrix_Pop(void);
+void Matrix_Get(MtxF* dest);
+void Matrix_Put(MtxF* src);
+MtxF* Matrix_GetCurrent(void);
+
+/* Basic operations */
+
+void Matrix_Mult(MtxF* mf, u8 mode);
+void Matrix_Translate(f32 x, f32 y, f32 z, u8 mode);
+void Matrix_Scale(f32 x, f32 y, f32 z, u8 mode);
+void Matrix_RotateX(f32 x, u8 mode);
+void Matrix_RotateY(f32 y, u8 mode);
+void Matrix_RotateZ(f32 z, u8 mode);
+
+/* Compound operations */
+
+void Matrix_RotateZYX(s16 x, s16 y, s16 z, u8 mode);
+void Matrix_TranslateRotateZYX(Vec3f* translation, Vec3s* rotation);
+void Matrix_SetTranslateRotateYXZ(f32 translateX, f32 translateY, f32 translateZ, Vec3s* rot);
```

Comment:
it's not clear what makes these "compound" compared to others under "miscellaneous"

e.g.
- "compound" `Matrix_RotateZYX` vs "miscellaneous" `Matrix_RotateAxis` (both take a mode)
- "compound" `Matrix_SetTranslateRotateYXZ` vs "miscellaneous" `Matrix_SetTranslateScaleMtx2` (both set/overwrite the matrix)